mobile/lib/modules/search/models/search_filter.dart

```diff
@@ -0,0 +1,310 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+import 'dart:convert';
+
+import 'package:immich_mobile/shared/models/asset.dart';
+import 'package:openapi/api.dart';
+
+class SearchLocationFilter {
+  String? country;
+  String? state;
+  String? city;
+  SearchLocationFilter({
+    this.country,
+    this.state,
+    this.city,
+  });
+
+  SearchLocationFilter copyWith({
+    String? country,
+    String? state,
+    String? city,
+  }) {
+    return SearchLocationFilter(
+      country: country ?? this.country,
+      state: state ?? this.state,
+      city: city ?? this.city,
+    );
+  }
+
+  Map<String, dynamic> toMap() {
+    return <String, dynamic>{
+      'country': country,
+      'state': state,
+      'city': city,
+    };
+  }
+
+  factory SearchLocationFilter.fromMap(Map<String, dynamic> map) {
```

Comment:
Do we use/need to/fromMap and to/fromJson at all for the filter classes?
I think they are never used.

Reply:
No, they are not used. It is auto-generated. I use this extension on vscode to help generate these classes

https://marketplace.visualstudio.com/items?itemName=dotup.dart-data-class-generator